scripts/generate_hamilton_input_cfp.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+import re
+import sys
+
+from EPPs.common import GenerateHamiltonInputEPP, InvalidStepError
+
+
+class GenerateHamiltonInputCFP(GenerateHamiltonInputEPP):
+    """"Generate a CSV containing the necessary information to normalise the sample in a NRM plate"""
+    _use_load_config = False  # prevent the loading of the config
+    # Define the column headers that will be used in the Hamilton input file
+    csv_column_headers = ['Input Plate', 'Input Well', 'Output Plate', 'Output Well', 'DNA Volume',
+                          'RSB Barcode', 'RSB Volume']
+
+    # Define the output file
+    output_file_name = 'KAPA_MAKE_CFP.csv'
+
+    #Define the number of input containers that are permitted
+    _max_nb_input_containers = 9
+
+    #Define the number of output containers that are permitted
+    _max_nb_output_containers = 1
+
+    def _generate_csv_dict(self):
+        # csv_dict will be a dictionary that consists of the lines to be present in the Hamilton input file.
+        csv_dict = {}
+
+        # find the lot number, i.e. barcode, of the RSB reagent.
+        RSB_template = "LP[0-9]{7}-RSB"
+        reagent_lots = list(self.process.step.reagent_lots)
+
+        rsb_barcode = None
+        for lot in reagent_lots:
+            if re.match(RSB_template, lot.lot_number):
+                rsb_barcode = lot.lot_number
+
+        if not rsb_barcode:
+            raise InvalidStepError('Please assign RSB lot before generating Hamilton input.')
```

Comment:
Lines 28-38 are duplicated in generate_hamilton_input_imp_ssqc and generate_hamilton_input_ntp - should be merged

Reply:
New function added to common.py and duplicated lines removed from scripts